### PR TITLE
Bug 2317212: Fix Clients' list page filtering

### PR DESCRIPTION
**Pull request by SanjalKatiyar**
**Before:**
Screenshot 2024-10-21 at 12 46 35 PM

**After:**
Screenshot 2024-10-21 at 12 51 00 PM

### CODE REVIEW

File: packages/odf/components/storage-consumers/list-filter.ts [hunk 1]
```diff
@@ -75,3 +76,15 @@ export const versionMismatchFilter = (
     );
   },
 });
+
+export const nameFilterOverride = (): RowFilter<StorageConsumerKind> => ({
```
Comment (bipuladh) [inline packages/odf/components/storage-consumers/list-filter.ts:80]:
```suggestion
export const storageConsumerNameFilter = (): RowFilter<StorageConsumerKind> => ({
```

File: packages/odf/components/storage-consumers/client-list.tsx [hunk 2]
```diff
@@ -424,14 +425,23 @@ export const ClientListPage: React.FC<ClientListPageProps> = () => {
...
-  const rowFilters = React.useMemo(
-    () => [clientHeartBeatFilter(t), versionMismatchFilter(t, serviceVersion)],
-    [t, serviceVersion]
-  );
+  // "rowFilters":
+  // - custom filters (new) that we are introducing to the list page
+  // - passing to both "useListPageFilter" hook & "ListPageFilter" component
+  // "rowFiltersWithNameOverride":
+  // - needed for overriding the filtering (default) on the CR's "name" (need to read the name from the CR's status instead)
+  // - only passing to "useListPageFilter" hook & not "ListPageFilter" component
```
Comment (bipuladh) [inline packages/odf/components/storage-consumers/client-list.tsx:433]: Why not list page filter component?

Comment (SanjalKatiyar) [reply packages/odf/components/storage-consumers/client-list.tsx:433]: We have a separate dropdown for "Name" filtering on the UI, already managed by `ListPageFilter` component by default. All we need to do is to override filter's definition in `useListPageFilter` hook.

ListPageFilter + rowFilters adds a different/new dropdown, passing `rowFiltersWithNameOverride` to `ListPageFilter` is adding extra empty item to that dropdown.

**Review by bipuladh — APPROVED** [review]
/lgtm

**Pull request merged**